libp2p/protocols/protocol.nim

```diff
@@ -19,14 +19,16 @@ const
 
 type
   LPProtoHandler* = proc (
-    conn: Connection,
-    proto: string):
-    Future[void]
-    {.gcsafe, raises: [].}
+      conn: Connection,
+      proto: string): Future[void] {.async.}
+
+  LPProtoHandler2*[E] = proc (
+      conn: Connection,
+      proto: string): InternalRaisesFuture[void, E]
 
   LPProtocol* = ref object of RootObj
     codecs*: seq[string]
-    handler*: LPProtoHandler ## this handler gets invoked by the protocol negotiator
+    handlerImpl: LPProtoHandler  ## invoked by the protocol negotiator
```

Comment:
Yes, that's what I mentioned above. It has to be renamed otherwise `handler=` doesn't take precedence, and `handler=` is used a lot, while using `LPProtocol(...)` over `LPProtocol.new` was only used in 1 example and 1 test, where it seems to be not intentional.

Reply:
breaks nimbus-eth2 though

Reply:
1 liner to fix